syntax = "proto3";

package user.v1;

message User {
  int32 id = 1;
  string first_name = 2;
  string last_name = 3;
  string email = 4;
}

message GetUserRequest {
  int32 id = 1;
}

message GetUserResponse {
  User user = 1;
}

message GetAllUsersRequest {}

message GetAllUsersResponse {
    repeated User users = 1;
}

service UserService {
  rpc GetUserById(GetUserRequest) returns (GetUserResponse);
  rpc GetAllUsers(GetAllUsersRequest) returns (GetAllUsersResponse);
}


